tests/cancun/eip1153_tstore/test_tstorage_clear_after_tx.py

```diff
@@ -0,0 +1,118 @@
+"""
+Ethereum Transient Storage EIP Tests
+https://eips.ethereum.org/EIPS/eip-1153
+"""
+
+from typing import Optional
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Bytecode,
+    Environment,
+    EVMCodeType,
+    Initcode,
+    Transaction,
+)
+from ethereum_test_tools.eof.v1 import Container
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import ref_spec_1153
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_1153.git_path
+REFERENCE_SPEC_VERSION = ref_spec_1153.version
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.with_all_evm_code_types
+def test_tstore_clear_after_deployment_tx(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    evm_code_type: EVMCodeType,
+):
+    """
+    This test first creates a contract, which TSTOREs a value 1 in slot 1.
+    After creating the contract, a new tx will call this contract, storing TLOAD(1) into slot 1.
+    The transient storage should be cleared after creating the contract (at tx-level), so
+    the storage should stay empty.
+    """
+    env = Environment()
+
+    init_code = Op.TSTORE(1, 1)
+    deploy_code = Op.SSTORE(1, Op.TLOAD(1))
```

Comment:
```suggestion
    deploy_code = Op.SSTORE(1, Op.TLOAD(1)) + Op.STOP
```

Reply:
I was about to comment "we don't need `Op.STOP`", but this to make the EOF container valid - right?

Reply:
Yes that is correct, only because of EOF.